### PR TITLE
feat: Solana Vault swaps (base PR)

### PR DESCRIPTION
**Pull request by albert-llimos**
# Pull Request

Closes: PRO-1630, PRO-1631

## Checklist

Please conduct a thorough self-review before opening the PR.

- [x] I am confident that the code works.
- [ ] I have written sufficient tests.
- [ ] I have written and tested required migrations.
- [ ] I have updated documentation where appropriate.

## Summary

~~Missing the migration to add the new variables into the environment.~~ Done

### CODE REVIEW

**Review by martin-chainflip — APPROVED** [review]

**Comment by albert-llimos** [comment]
We will create PRs on top of this one for the Solana Program Swap feature. Therefore I'd not merge this to main and merge the other PRs onto this one. Once the feature is complete we can then merge it to main.

File: state-chain/pallets/cf-broadcast/src/migrations.rs
Comment (ramizhasan111) [inline state-chain/pallets/cf-broadcast/src/migrations.rs:10]: why this change?

File: state-chain/chains/src/sol/transaction_builder.rs
Comment (ramizhasan111) [inline state-chain/chains/src/sol/transaction_builder.rs:462]: i think you dont need the extra map. you could do this inside the ```flat_map```:
```rust
.flat_map(|(event_account, payee)| vec![AccountMeta::new(event_account.into(), false), AccountMeta::new(payee.into(), false)])
```

File: state-chain/runtime/src/migrations/sol_api_environment.rs
Comment (ramizhasan111) [inline state-chain/runtime/src/migrations/sol_api_environment.rs:1]: we dont need this migration to be implemented here at the top level of the runtime. We can move it into the pallet level migration.

File: state-chain/pallets/cf-broadcast/src/migrations.rs
Comment (albert-llimos) [reply state-chain/pallets/cf-broadcast/src/migrations.rs:10]: @syan095

File: state-chain/runtime/src/migrations/sol_api_environment.rs
Comment (albert-llimos) [reply state-chain/runtime/src/migrations/sol_api_environment.rs:1]: @syan095

File: state-chain/pallets/cf-broadcast/src/migrations.rs
Comment (syan095) [reply state-chain/pallets/cf-broadcast/src/migrations.rs:10]: We have `SerializeSolanaBroadcastMigration` in the Runtime crate that migrates broadcaster pallets from version 8 -> 9, so I've updated the "placeholder" migration (which should always points to the current version).

I've updated this with a comment to make it clearer

Comment (syan095) [reply state-chain/pallets/cf-broadcast/src/migrations.rs:10]: I see. I was under the impression that the placeholder migration should point to the latest version after update, but seems like we don't need to. 

This change was overridden by a merge from main.

File: state-chain/runtime/src/migrations/sol_api_environment.rs
Comment (syan095) [reply state-chain/runtime/src/migrations/sol_api_environment.rs]: Got it. Moved it into the Environment pallet.

File: state-chain/pallets/cf-broadcast/src/migrations.rs
Comment (syan095) [reply state-chain/pallets/cf-broadcast/src/migrations.rs]: UPDATE: the placeholder version should match the latest pallet's storage version, or it will give out a warning on try-runtime.
I've updated this to version 10 (as 10 is the latest version after merge from main)

Comment (syan095) [reply state-chain/pallets/cf-broadcast/src/migrations.rs:11]: Update again: Looks like with version 10, we get a warning when doing the pallet migrations, but we are OK in the post-migration checks. 
If we leave the version at 8, we are OK when doing pallet migrations, but will get a warning in post-migration checks.

this is due to how we do the migrations - we do pallet migrations first, then the Runtime ones.

It's just warnings, so I guess I'll just leave it at version 10, since that's the version the pallets ended up on after all the migrations are done.

**Review by syan095 — APPROVED** [review]

**Pull request merged**
